easybuild/framework/easyconfig/style.py

```diff
@@ -0,0 +1,128 @@
+# definied as WXXX and EXXX (for warnings and errors) where XXX is a
+# 3 digit number.
+#
+# It should be mentioned in the docstring as a single word.
+# Read the pep8 docs to understand the arguments of these functions:
+# https://pep8.readthedocs.org or more specifically:
+# https://pep8.readthedocs.org/en/latest/developer.html#contribute
+def _eb_check_trailing_whitespace(physical_line, lines, line_number, total_lines):  # pylint:disable=unused-argument
+    """
+    W299
+    Warn about trailing whitespace, except for the description and comments.
+    This differs from the standard trailing whitespace check as that
+    will warn for any trailing whitespace.
+    The arguments are explained at
+    https://pep8.readthedocs.org/en/latest/developer.html#contribute
+    """
+    comment_re = re.compile(r'^\s*#')
+    if comment_re.match(physical_line):
+        return None
+
+    result = pep8.trailing_whitespace(physical_line)
+    if result:
+        result = (result[0], result[1].replace("W291", "W299"))
+
+    # if the warning is about the multiline string of description
+    # we will not issue a warning
+    keys_re = re.compile(r"^(?P<key>[a-z_]+)\s*=\s*")
+
+    # starting from the current line and going to the top,
+    # check that if the first `key = value` that is found, has
+    # key == description, then let the test pass, else return
+    # the result of the general pep8 check.
+    for line in reversed(lines[:line_number]):
+        res = keys_re.match(line)
+        if res:
+            if res.group("key") == "description":
+                return None
+            else:
+                break
+
+    return result
+
+
+@only_if_module_is_available('pep8')
+def style_conformance(easyconfigs, verbose=False):
+    """
+    Check the given list of easyconfigs for style
+    :param easyconfigs list of file paths to easyconfigs
+    :param verbose print our statistics and be verbose about the errors and warning
+    :return the number of warnings and errors
+    """
+    # importing autopep8 changes some pep8 functions.
+    # We reload it to be sure to get the real pep8 functions.
+    reload(pep8)
+
+    # register the extra checks before using pep8:
+    # any function in this module starting with `_eb_check_` will be used.
+    cands = globals()
+    for check_function in sorted([cands[f] for f in cands if callable(cands[f]) and f.startswith('_eb_check_')]):
+        _log.debug("Adding custom style check %s", check_function)
+        pep8.register_check(check_function)
+
+    pep8style = pep8.StyleGuide(quiet=False, config_file=None)
+    options = pep8style.options
+    # we deviate from standard pep8 and allow 120 chars
+    # on a line: the default of 79 is too narrow.
+    options.max_line_length = 120
+    # we ignore some tests
+    # note that W291 has be replaced by our custom W299
+    options.ignore = (
+        'E402',  # import not on top
+        'W291',  # replaced by W299
+        'E501',  # line too long
```

Comment:
Why are we ignoring E402 and E501? If there's a good reason, please mention why.